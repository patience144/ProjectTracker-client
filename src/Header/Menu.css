.header-options {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-self: center;
  align-items: center;
  width: 50%;
  height: 50px;
  margin-bottom: 10px;
}

.header-options a,
.header-options a {
  color: black;
  text-decoration: none;
  font-size: 25px;
  font-weight: bold;
  font-family: "Cormorant Garamond", serif;
}

.header-options a {
  background-color: transparent;
  border: none;
  cursor: pointer;
}

.buttons {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  width: 110%;
  margin: 0;
}

.header-options h4 {
  font-size: 2.5vh;
}

.loggedIn {
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 100px;
}

.loggedIn button {
  margin-bottom: 50px;
  margin-left: 25px;
  border: solid;
  color: black;
}

#users-button,
#account-button {
  margin-right: 15px;
}

.nav li + li {
  margin-top: 25%;
}

header h1 {
  font-size: 4vh;
}

header h4 {
  font-size: 2vh;
}

@media all and (max-width: 400px) {
  .header-options {
    align-self: flex-end;
    box-sizing: border-box;
    height: 50px;
    width: 65%;
  }

  .header-options a,
  .header-options a {
    font-size: 15px;
    margin: 0;
  }

  .buttons {
    align-items: flex-end;
  }

  .header-options h4 {
    font-size: 5.8vh;
  }
}
