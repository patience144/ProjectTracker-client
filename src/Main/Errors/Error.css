.error {
  align-self: center;
}

h3, p {
  color: white;
}