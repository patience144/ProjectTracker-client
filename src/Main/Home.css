.home {
  height: 100%;
  z-index: 0;
}

main {
  width: 70%;
}

h2 {
  color: white;
  z-index: 1;
}

.home p {
  text-align: left;
  padding: 0 15px;
  z-index: 1;
}

ul {
  color: white;
  z-index: 1;
}

.home-secondary p {
  text-align: center;
  margin-bottom: 8px;
}

img {
  opacity: 0.3;
  position: fixed;
  top: 150px;
  left: 0;
  z-index: 0;
}

.home-secondary {
  margin-top: 5%;
}

.home a {
  background-color: rgba(163, 22, 219, 0.925);
  text-decoration: none;
  font-weight: bold;
  font-size: 18px;
  color: rgb(46, 30, 75);
  align-self: center;
  border: solid;
  border-color: rgb(121, 22, 151);
  border-width: 1px;
  border-radius: 3px;
  width: fit-content;
  margin-top: 25px;
  padding: 15px 35px;
  z-index: 0;
}

.home a:hover {
  background-color: rgba(238, 56, 245, 0.699);
}

.home:active a {
  transform: scale(0.95);
}

@media all and (min-width: 400px) {
  .home {
    font-size: 125%;
    width: 70%;
  }

  .home p,
  ul {
    padding: 0 10%;
  }

  ul {
    align-self: center;
  }

  img {
    opacity: 0.11;
  }
}
