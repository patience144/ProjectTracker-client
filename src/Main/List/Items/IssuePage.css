.issue-page {
  display: flex;
  flex-direction: column;
  width: 40%;
  color: white;
}

.issue-page h3 {
  text-align: center;
  margin-top: 15%;
}

.issue-page label {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  box-sizing: border-box;
  display: flex;
  margin-top: 25px;
  width: 100%;
}

.issue-page input {
  min-width: 50%;
}

.issue-buttons {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.issue-buttons button,
.issue-page button {
  background-color: rgba(163, 22, 219, 0.925);
  border: none;
  min-width: 100px;
  height: 1.8em;
  outline: none;
  font-family: "Dancing Script", cursive;
  font-weight: bolder;
  font-size: 20px;
  color: rgb(95, 0, 95);
  border-radius: 3px;
  margin: 0 20px;
  cursor: pointer;
}

.issue-page button:hover {
  background-color: rgba(238, 56, 245, 0.699);
  border-color: rgba(238, 56, 245, 0.699);
}

.issue-buttons button:active,
.issue-page button:active {
  transform: scale(0.92);
}

.issue-page button {
  margin-top: 25px;
}

@media all and (max-width: 400px) {
  .issue-page {
    width: 75%;
  }

  .issue-buttons button {
    margin: 0 10px;
  }
}
