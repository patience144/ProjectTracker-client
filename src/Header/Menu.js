import React from "react";
import { Link, withRouter } from "react-router-dom";
import { API_KEY } from "../config";
import jwt_decode from "jwt-decode";
import UserContext from "../context/UserContext";
import "./Menu.css";

class Menu extends React.Component {
  state = { menu: false };

  static contextType = UserContext;

  render() {
    const menu = this.state.menu;
    const token = window.sessionStorage.getItem("authToken");
    const id = this.context ? this.context.id : null;
    let username = token ? this.context.username : null;
    const admin = username === "admin";
    if (!username && token && token !== API_KEY)
      username = jwt_decode(token).sub;
    return (
      <>
      {(username) ? <h4>Welcome, {username}.</h4> : null}
      <nav className={menu ? "faded" : ""}>
        <ul>
          <li><Link to="/signup">Sign-Up</Link></li>
          <li><Link to="/login">Login</Link></li>
          <li>
            <a
              href="https://github.com/patience144/projecttracker-client"
              target="_blank"
              rel="noreferrer"
            >
              GitHub
            </a>
          </li>
        </ul>
        <i
          onClick={(evt) => {
            evt.stopPropagation();
            this.setState({ menu: !menu });
          }}
          onBlur={this.blur}
          className="fa fa-bars"
        ></i>
        <div className={`nav ${menu ? "show-menu" : ""}`}>
          <ul>
            {
              // Show different access buttons depending if Logged in or not.
              !token ? (
                <>
                  <li>
                    <Link to="/signup">Sign-Up</Link>
                  </li>
                  <li>
                    <Link to="/login">Login</Link>
                  </li>
                </>
              ) : (
                admin
                  ? <li><Link id="users-button" to={`/users`}>Users</Link></li>
                  : <li><Link id="account-button" to={`/users/${id}`}>Accounts</Link></li>
              )
            }
            <li>
              <a
                href="https://github.com/patience144/projecttracker"
                target="_blank"
                rel="noreferrer"
              >
                GitHub
              </a>
            </li>
            {
              (!token)
                ? null
                : <li>
                    <Link
                      to="/logout"
                      id="logout-button"
                      onClick={() => {
                        window.sessionStorage.removeItem("authToken");
                        this.props.history.push("/");
                      }}
                    >
                      Log Out
                    </Link>
                  </li>
            }
          </ul>
        </div>
      </nav>
      </>
    );
  }

  componentDidMount() {
    document.addEventListener("click", () => this.setState({ menu: false }));
  }
}

export default withRouter(Menu);
