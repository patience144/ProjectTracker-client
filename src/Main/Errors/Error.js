import React from 'react';
import PropTypes from 'prop-types';
import './Error.css';

class Error extends React.Component {
  static defaultProps = { message: '' }

  static propTypes = { message: PropTypes.string };
  
  render() {
    return (
      <div className='error'>
        <h3>ERROR</h3>
        {/* Different messages passed as props from error handling in components*/}
        <p>{this.props.message}</p>
      </div>
    );
  };
};

export default Error;