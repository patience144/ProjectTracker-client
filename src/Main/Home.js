import React from 'react';
import { Link } from 'react-router-dom';
import background from '../img/background2.png';
import './Home.css';

class Home extends React.Component {
  render() {
    return (
      <div className='home'>
        <div className='home-primary'>
          <img src={background} alt='background' />
          <h2>Welcome to ProjectTracker!</h2>
          <p>A stopping point for developers to keep keep track of their projects and bugs.</p>
          <p>Establish a phase and status for each project and issue, so you can be thorough on every project's health!</p>
          <Link to='projects'>Projects</Link>
          <a href='https://github.com/patience144/projecttracker-client'>GitHub</a></div>
        <div className='home-secondary'>
          <p>It's easy:</p>
          <ul>
            <li>Sign-Up.</li>
            <li>Create a project.</li>
            <li>Establish your issues.</li>
            <li>Share away!</li>
          </ul>
          <p><i>(Next up: Implementing collaborators and teams!)</i></p>
        </div>
      </div>
    );
  };
};

export default Home;