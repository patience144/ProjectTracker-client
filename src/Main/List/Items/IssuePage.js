import React from 'react';
import { withRouter } from 'react-router-dom';
import jwt_decode from 'jwt-decode';
import PropTypes from 'prop-types';
import { API_KEY } from '../../../config';
import './IssuePage.css';

class IssuePage extends React.Component {
  static defaultProps = { state: {} };

  static propTypes = {state: PropTypes.object.isRequired};

  render() {
    const token = window.sessionStorage.getItem('authToken');
    const admin = token === API_KEY;
    const username = (token && !admin) ? jwt_decode(token).sub : null;
    const issueID = this.props.match.params.issueID;
    const issue = 
      (this.props.state.issues)
        ? this.props.state.issues.find(issue => issue.id === issueID)
        : {};
    const startDate = new Date(issue.startDate).toDateString().slice(4);
    const project = 
    (this.props.state.projects)
      ? this.props.state.projects.find(project => project.id === this.props.match.params.projectID)
      : {};
    const projectName = (project) ? project.name : null;
    return (
      <form onSubmit={this.handleSave} className='issue-page'>
        <h3>{issue.name || 'New Issue'}</h3>
        <label>Project: {projectName}</label>
        <label>Name: {issue.name}</label>
        <label>Description: {issue.description}</label>
        <label>Owner: {issue.owner}</label>
        <label>Languages/Tools: {issue.tools}</label>
        <label>Phase: {issue.phase}</label>
        <label>Status: {issue.status}</label>
        <label>Start Date: {startDate}</label>
        <label>Collaboration: {(issue.collaboration) ? issue.collaboration.toString() :  null}</label>
        <label>GitHub: {issue.github}</label>
        <div className='issue-buttons'>
        {
          // If not a New Project, show 'Issues' button to browse project's issues.
            (token && (admin || username === issue.owner))
              ? <>
                  <button type='button' onClick={() => {
                    return this.props.history.push(`/edit/projects/${issue.projectID}/issues/${issueID}`)
                  }}>Edit</button>
                </>
              : null
          }
          <button type='button' onClick={this.props.history.goBack}>Back</button>
        </div>
      </form>
    );
  };
};

export default withRouter(IssuePage);