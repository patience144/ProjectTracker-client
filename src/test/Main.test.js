import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import List from '../Main/List/List';
import Item from '../Main/List/Items/Item';
import ProjectPage from '../Main/List/Items/ProjectPage';
import IssuePage from '../Main/List/Items/IssuePage';
import Error from '../Main/Errors/Error';
import UserPage from '../Main/Users/UserPage';
import Home from '../Main/Home';
import Menu from '../Header/Menu';

describe('Main Section', () => {
  const list = <BrowserRouter><List /></BrowserRouter>;
  const item = <BrowserRouter><Item /></BrowserRouter>;
  const error = <BrowserRouter><Error /></BrowserRouter>;
  const project = <BrowserRouter><ProjectPage /></BrowserRouter>;
  const issue = <BrowserRouter><IssuePage /></BrowserRouter>;
  const user = <BrowserRouter><UserPage /></BrowserRouter>;
  const menu = <BrowserRouter><Menu /></BrowserRouter>;
  const home = <BrowserRouter><Home /></BrowserRouter>
  describe('The List', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(list, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
      const tree = renderer.create(list).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Items', () => {
    describe('The Item component', () => {
      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(item, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('render the UI as expected', () => {
        const tree = renderer.create(item).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The Project Page', () => {
      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(project, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(project).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The Issue Page', () => {
      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(issue, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(issue).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
  describe('The Errors', () => {
    it('render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(error, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('render the UI as expected', () => {
      const tree = renderer.create(error).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Users', () => {
    it('render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(user, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('render the UI as expected', () => {
      const tree = renderer.create(user).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Menu', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(menu, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('render the UI as expected', () => {
      const tree = renderer.create(menu).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Home Page', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(home, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('render the UI as expected', () => {
      const tree = renderer.create(home).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});