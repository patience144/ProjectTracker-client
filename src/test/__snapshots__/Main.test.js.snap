// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Section The Errors render the UI as expected 1`] = `
<div
  className="error"
>
  <h3>
    ERROR
  </h3>
  <p>
    
  </p>
</div>
`;

exports[`Main Section The Home Page render the UI as expected 1`] = `
<div
  className="home"
>
  <div
    className="home-primary"
  >
    <img
      alt="background"
      src="background2.png"
    />
    <h2>
      Welcome to DevMap!
    </h2>
    <p>
      A stopping point for developers to keep keep track of their projects and bugs.
    </p>
    <p>
      Establish a phase and status for each project and issue, so you can be thorough on every project's health!
    </p>
    <a
      href="/projects"
      onClick={[Function]}
    >
      Projects
    </a>
  </div>
  <div
    className="home-secondary"
  >
    <p>
      It's easy:
    </p>
    <ul>
      <li>
        Sign-Up.
      </li>
      <li>
        Create a project.
      </li>
      <li>
        Establish your issues.
      </li>
      <li>
        Share away!
      </li>
    </ul>
    <p>
      <i>
        (Next up: Implementing collaborators and teams!)
      </i>
    </p>
  </div>
</div>
`;

exports[`Main Section The Items The Issue Page renders the UI as expected 1`] = `
<form
  className="issue-page"
>
  <h3>
    New Issue
  </h3>
  <label>
    Project: 
  </label>
  <label>
    Name: 
  </label>
  <label>
    Description: 
  </label>
  <label>
    Owner: 
  </label>
  <label>
    Languages/Tools: 
  </label>
  <label>
    Phase: 
  </label>
  <label>
    Status: 
  </label>
  <label>
    Start Date: 
    lid Date
  </label>
  <label>
    Collaboration: 
  </label>
  <label>
    GitHub: 
  </label>
  <div
    className="issue-buttons"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
  </div>
</form>
`;

exports[`Main Section The Items The Item component render the UI as expected 1`] = `
<a
  className="item"
  href="/projects/undefined"
  onClick={[Function]}
>
  <div
    className="item-main"
  >
    <label />
    <p />
    <div
      className="item-health"
    >
      <label />
      <label />
    </div>
  </div>
  <div
    className="item-tags"
  >
    <label />
  </div>
</a>
`;

exports[`Main Section The Items The Project Page renders the UI as expected 1`] = `
<form
  className="project-page"
>
  <h3>
    New Project
  </h3>
  <label>
    Name: 
  </label>
  <label>
    Description: 
  </label>
  <label>
    Languages/Tools: 
  </label>
  <label>
    Phase: 
  </label>
  <label>
    Status: 
  </label>
  <label>
    Start Date: 
    lid Date
  </label>
  <label>
    Owner: 
  </label>
  <label>
    Collaboration: 
  </label>
  <label>
    GitHub: 
  </label>
  <div
    className="project-buttons"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
  </div>
</form>
`;

exports[`Main Section The List renders the UI as expected 1`] = `
Array [
  <h3>
    All Projects
  </h3>,
  <a
    className="create"
    href="/new-project"
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    New Project
  </a>,
  <div
    className="item-list"
  />,
]
`;

exports[`Main Section The Menu render the UI as expected 1`] = `
<div
  className="header-options"
>
  <div
    className="buttons"
  >
    <a
      href="/signup"
      onClick={[Function]}
    >
      Sign-Up
    </a>
    <a
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
</div>
`;

exports[`Main Section The Users render the UI as expected 1`] = `
<form
  className="user-page"
>
  <h3>
    Profile
  </h3>
  <label>
    Username: 
  </label>
  <label>
    First Name: 
  </label>
  <label>
    Last Name: 
  </label>
  <label>
    E-mail: 
  </label>
  <label>
    Tools: 
  </label>
  <label>
    Start Date: 
  </label>
  <label>
    GitHub: 
  </label>
  <div
    className="issue-buttons"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Edit
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
  </div>
</form>
`;
